sysdig/resource_sysdig_secure_cloud_auth_account.go

```diff
@@ -650,6 +656,50 @@ func componentsToResourceData(components []*cloudauth.AccountComponent, dataComp
 
 					componentBlock[SchemaServicePrincipalMetadata] = string(schema)
 				}
+
+				if providerKey, ok := provider.(*cloudauth.ServicePrincipalMetadata_Azure_); ok {
+					jsonifiedKey := struct {
+						AccountEnabled         bool   `json:"account_enabled"`
+						AppDisplayName         string `json:"app_display_name"`
+						AppId                  string `json:"app_id"`
+						AppOwnerOrganizationId string `json:"app_owner_organization_id"`
+						DisplayName            string `json:"display_name"`
+						Id                     string `json:"id"`
+					}{
+						AccountEnabled:         providerKey.Azure.GetActiveDirectoryServicePrincipal().GetAccountEnabled(),
+						AppDisplayName:         providerKey.Azure.GetActiveDirectoryServicePrincipal().GetAppDisplayName(),
+						AppId:                  providerKey.Azure.GetActiveDirectoryServicePrincipal().GetAppId(),
+						AppOwnerOrganizationId: providerKey.Azure.GetActiveDirectoryServicePrincipal().GetAppOwnerOrganizationId(),
+						DisplayName:            providerKey.Azure.GetActiveDirectoryServicePrincipal().GetDisplayName(),
+						Id:                     providerKey.Azure.GetActiveDirectoryServicePrincipal().GetId(),
+					}
+
+					bytesKey, err := json.Marshal(jsonifiedKey)
+					if err != nil {
+						fmt.Printf("Failed to populate %s: %v", SchemaServicePrincipalMetadata, err)
+						break
+					}
+
+					// update the json with proper indentation
+					var out bytes.Buffer
+					if err := json.Indent(&out, bytesKey, "", "  "); err != nil {
+						fmt.Printf("Failed to populate %s: %v", SchemaServicePrincipalMetadata, err)
+						break
+					}
+					out.WriteByte('\n')
+
+					schema, err := json.Marshal(map[string]interface{}{
+						"azure": map[string]interface{}{
+							"key": out.Bytes(),
```

Comment:
I believe the structure here is different for azure? instead of `key` it will be `active_directory_service_principal`

Reply:
Good catch!